### ISSUE TITLE
Add `no-useless-string-literal`

### ISSUE THREAD

**Issue by ota-meshi** [ota-meshi/eslint-plugin-regexp#638]
**Description**

`no-useless-string-literal` rule reports the string alternatives of a single character.
They can be placed outside `\q{...}`.

```js
/* ✗ BAD */
var foo = /[\q{a}]/v // => /[a]/v
var foo = /[\q{a|bc}]/v // => /[a\q{bc}]/v

/* ✓ GOOD */
var foo = /[\q{ab|}]/v
```
